tempodb/encoding/vparquet/block_search.go

```diff
@@ -73,71 +84,172 @@ func (b *backendBlock) Search(ctx context.Context, req *tempopb.SearchRequest, o
-			traceIters = append(traceIters, makeIter("RootSpanName", pq.NewSubstringPredicate(v), ""))
-		case LabelServiceName:
-			resourceIters = append(resourceIters, makeIter("rs.Resource.ServiceName", pq.NewSubstringPredicate(v), ""))
-		case LabelCluster:
-			resourceIters = append(resourceIters, makeIter("rs.Resource.Cluster", pq.NewSubstringPredicate(v), ""))
-		case LabelNamespace:
-			resourceIters = append(resourceIters, makeIter("rs.Resource.Namespace", pq.NewSubstringPredicate(v), ""))
-		case LabelPod:
-			resourceIters = append(resourceIters, makeIter("rs.Resource.Pod", pq.NewSubstringPredicate(v), ""))
-		case LabelContainer:
-			resourceIters = append(resourceIters, makeIter("rs.Resource.Container", pq.NewSubstringPredicate(v), ""))
-		case LabelK8sClusterName:
-			resourceIters = append(resourceIters, makeIter("rs.Resource.K8sClusterName", pq.NewSubstringPredicate(v), ""))
-		case LabelK8sNamespaceName:
-			resourceIters = append(resourceIters, makeIter("rs.Resource.K8sNamespaceName", pq.NewSubstringPredicate(v), ""))
-		case LabelK8sPodName:
-			resourceIters = append(resourceIters, makeIter("rs.Resource.K8sPodName", pq.NewSubstringPredicate(v), ""))
-		case LabelK8sContainerName:
-			resourceIters = append(resourceIters, makeIter("rs.Resource.K8sContainerName", pq.NewSubstringPredicate(v), ""))
-		case LabelName:
-			resourceIters = append(resourceIters, makeIter("rs.ils.Spans.Name", pq.NewSubstringPredicate(v), ""))
-		case LabelHTTPMethod:
-			resourceIters = append(resourceIters, makeIter("rs.ils.Spans.HttpMethod", pq.NewSubstringPredicate(v), ""))
-		case LabelHTTPUrl:
-			resourceIters = append(resourceIters, makeIter("rs.ils.Spans.HttpUrl", pq.NewSubstringPredicate(v), ""))
-		case LabelHTTPStatusCode:
+		column := labelMappings[k]
+
+		// if we don't have a column mapping then pass it forward to otherAttribute handling
+		if column == "" {
+			otherAttrConditions[k] = v
+			continue
+		}
+
+		// most columns are just a substring predicate over the column, but we have
+		// special handling for http status code and span status
+		if k == LabelHTTPStatusCode {
 			if i, err := strconv.Atoi(v); err == nil {
-				resourceIters = append(resourceIters, makeIter("rs.ils.Spans.HttpStatusCode", pq.NewIntBetweenPredicate(int64(i), int64(i)), ""))
+				resourceIters = append(resourceIters, makeIter(column, pq.NewIntBetweenPredicate(int64(i), int64(i)), ""))
 				break
 			}
 			// Non-numeric string field
 			otherAttrConditions[k] = v
-		case StatusCodeTag:
+			continue
+		}
+		if k == LabelStatusCode {
 			code := StatusCodeMapping[v]
-			resourceIters = append(resourceIters, makeIter("rs.ils.Spans.StatusCode", pq.NewIntBetweenPredicate(int64(code), int64(code)), ""))
-		default:
-			otherAttrConditions[k] = v
+			resourceIters = append(resourceIters, makeIter(column, pq.NewIntBetweenPredicate(int64(code), int64(code)), ""))
+			continue
 		}
+
+		traceIters = append(traceIters, makeIter(column, pq.NewSubstringPredicate(v), ""))
```

Comment:
Previously, only `RootServiceName` and `RootSpanName` were trace-level iters.  Trying to think how this changes the search behavior... it would have meant that all of the resource-level conditions were asserted on the same batch, but now can match any batch anywhere in the trace.  Thoughts?

Reply:
good point. Added a check to only add the two root attributes to traceIters and the rest to resourceIters